### PR TITLE
Adds AltCurrencyPaymentWithDestinationTag struct

### PR DESCRIPTION
**Pull request by JasonTulp**
Adds AltCurrencyPaymentWithDestinationTag struct to handle non XRP bridge transactions that have a specified DestinationTag

### CODE REVIEW

File: codec/tests/transaction_decoding.rs [hunk 1]
```diff
@@ -735,7 +892,7 @@ fn decode_SignerListSet_tx_empty_signer_entries() {
 }
 
 #[test]
-fn serialize_NFTokenCreateOffer_tx() {
+fn serialize_nftoken_create_offer_tx() {
```
Comment (JasonTulp) [inline codec/tests/transaction_decoding.rs:895]: This change was to remove some warnings

File: codec/tests/transaction_decoding.rs [hunk 2]
```diff
@@ -17,11 +15,13 @@ use xrpl_codec::{
 
 // Assert `encoded` input decodes to `expected` JSON format (whitespace will be removed)
 fn assert_decodes(encoded: &[u8], expected: &str) {
+    println!("{:?}", encoded);
```
Comment (surangap) [inline codec/tests/transaction_decoding.rs:18]: nit
```suggestion
```

**Review by surangap — APPROVED** [review]
lgtm

File: codec/tests/transaction_decoding.rs (hunk 2 above)
Comment (JasonTulp) [reply codec/tests/transaction_decoding.rs]: https://github.com/futureversecom/xrpl-tx-codec/pull/21/commits/e8c79ec4f114dadcccef9018b5d26b3c169d1a74

**Pull request merged**